src/geoclaw_riemann_utils_vec.f90

```diff
@@ -0,0 +1,698 @@
+! This code has been adapted for vectorization. 
+! For more details on how to use it, see the Makefile of the chile2010 example.
+
+!-----------------------------------------------------------------------
+      subroutine riemann_aug_JCP(maxiter,meqn,mwaves,hL,hR,huL,huR, &
+        hvL,hvR,bL,bR,uL,uR,vL,vR,phiL,phiR,pL,pR,sE1,sE2,drytol,g,rho, &
+        sw1,sw2,sw3,fw11,fw12,fw13,fw21,fw22,fw23,fw31,fw32,fw33)
+      !dir$ attributes forceinline :: riemann_aug_JCP
+      !dir$ attributes vector: uniform(maxiter,meqn,mwaves,drytol,g,rho) :: riemann_aug_JCP
+
+      ! solve shallow water equations given single left and right states
+      ! This solver is described in J. Comput. Phys. (6): 3089-3113, March 2008
+      ! Augmented Riemann Solvers for the Shallow Equations with Steady States and Inundation
+
+      ! To use the original solver call with maxiter=1.
+
+      ! This solver allows iteration when maxiter > 1. The iteration seems to help with
+      ! instabilities that arise (with any solver) as flow becomes transcritical over variable topo
+      ! due to loss of hyperbolicity.
+
+      implicit none
+
+      !input
+      integer meqn,mwaves,maxiter
+      !double precision fw(meqn,mwaves)
+      !double precision sw(mwaves)
+      double precision sw1,sw2,sw3,fw11,fw12,fw13,fw21,fw22,fw23,fw31,fw32,fw33
```

Comment:
We have been using `real(kind=8)` instead of `double precision` with the intent to perhaps support multiple precisions depending on the case.  I know the former is less portable but I think it's probably best to be consistent unless there is another reason to use `double precision`.

Reply:
I've now replaced double precision with real(kind=8) in my code. Note however that the original geoclaw_riemann_utils.f still uses double precision, you may want to check whether this should also be modified.